lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -35,15 +38,50 @@ import { replaceWindowsNetworkDriveLetter } from '../../os/windows-network-drive
 import {
 	ChangeButton,
 	DetailsText,
-	Footer,
 	StepButton,
 	StepNameButton,
 	StepSelection,
-	Underline,
 } from '../../styled-components';
+import { colors } from '../../theme';
 import { middleEllipsis } from '../../utils/middle-ellipsis';
 import { SVGIcon } from '../svg-icon/svg-icon';
 
+const jsonStorage = {
+	get: (key: string) => {
+		return new Promise((resolve, reject) => {
+			jsonStorageCb.get(key, (err, value) => {
+				if (err) {
+					reject(err);
+					throw err;
+				}
+				resolve(value);
+				return value;
+			});
+		});
+	},
+	set: (key: string, value: object) => {
+		return new Promise((resolve, reject) => {
+			jsonStorageCb.set(key, value, err => {
+				if (err) {
+					reject(err);
+					throw err;
+				}
+				resolve(value);
+				return value;
+			});
+		});
+	},
+};
+
+const getRecentUrlImages = () =>
+	jsonStorage.get('recentUrlImages') as Promise<string[]>;
```

Comment:
Please make `'recentUrlImages'` a constant and add an equivalent `setRecentUrlImages` function using it.
Check that `recentUrlImages` is an array and return a default empty array in `getRecentUrlImages`